x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/TextExpansionQueryIT.java

```diff
@@ -128,20 +127,15 @@ public void testTextExpansionQuery() throws IOException {
             "washing machine smells"
         );
 
-        List<List<Map<String, Float>>> tokenWeights = new ArrayList<>();
         // Generate the rank feature weights via the inference API
         // then index them for search
-        for (var input : inputs) {
-            Response inference = infer(input, modelId);
-            List<Map<String, Object>> responseMap = (List<Map<String, Object>>) entityAsMap(inference).get("inference_results");
-            Map<String, Object> inferenceResult = responseMap.get(0);
-            var idWeights = (List<Map<String, Float>>) inferenceResult.get("predicted_value");
-            tokenWeights.add(idWeights);
-        }
+        String pipelineId = putPipeline(modelId, "tokens");
 
         // index tokens
         createRankFeaturesIndex(indexName);
-        bulkIndexDocs(inputs, tokenWeights, indexName);
+        bulkIndexDocs(inputs, indexName, pipelineId);
+        var mapping = getIndexMapping(indexName);
+        var resp = responseAsMap(client().performRequest(new Request("GET", indexName + "/_search")));
```

Comment:
I am doing something dumb with this test. I am not sure why it is failing. Will have to come back to it :/

Or somebody from ML can take over.